x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/rank/textsimilarity/TextSimilarityRankRetrieverBuilder.java

```diff
@@ -32,7 +38,7 @@
 /**
  * A {@code RetrieverBuilder} for parsing and constructing a text similarity reranker retriever.
  */
-public class TextSimilarityRankRetrieverBuilder extends RetrieverBuilder {
+public class TextSimilarityRankRetrieverBuilder extends CompoundRetrieverBuilder<TextSimilarityRankRetrieverBuilder> {
 
     public static final NodeFeature TEXT_SIMILARITY_RERANKER_RETRIEVER_SUPPORTED = new NodeFeature(
         "text_similarity_reranker_retriever_supported"
```

Comment:
Should we introduce a new cluster feature as we did with `rrf` ?

Reply:
Added a new cluster feature to validate during parsing  in https://github.com/elastic/elasticsearch/pull/114085/commits/f90d278ce911119033e6957688381a8d342cbe0b